generate/generate.go

```diff
@@ -905,25 +905,31 @@ func (g *Generator) convertMap(parentName, fieldName string, mapTyp *types.Map)
 	return typeName, nestedType
 }
 
-func (g *Generator) convertParameter(tuple *types.Tuple) (*string, error) {
+func (g *Generator) convertParameter(tuple *types.Tuple) (*string, *bool, error) {
 	switch tuple.Len() {
 	case 0:
 		g.addProtoDep("google/protobuf/empty.proto")
-		return proto.String(".google.protobuf.Empty"), nil
+		return proto.String(".google.protobuf.Empty"), nil, nil
 	case 1:
 		// below
 	default:
-		return nil, fmt.Errorf("multiple parameters are not supported")
+		return nil, nil, fmt.Errorf("multiple parameters are not supported")
 	}
 	param := tuple.At(0).Type()
 	_, label, tname := g.convertType(param)
 	if tname == "" {
-		return nil, fmt.Errorf("unsupported parameter type: %v", param)
+		return nil, nil, fmt.Errorf("unsupported parameter type: %v", param)
 	}
 	if label == desc.FieldDescriptorProto_LABEL_REPEATED {
-		return nil, fmt.Errorf("parameter type should not be repeated")
+		return nil, nil, fmt.Errorf("parameter type should not be repeated")
 	}
-	return &tname, nil
+
+	isStream := proto.Bool(false)
+	if strings.Contains(param.String(), "chan") {
```

Comment:
don't stringify a type to see if it's a channel; this will break, for example, if one does `type chanFoo int`. instead, check the "type of the type" agains `types.Chan`.